qa/L0_perf_resnet/run_test.sh

```diff
@@ -73,20 +66,58 @@ fi
 
 set +e
 
-# Run the model once to warm up. Some frameworks do optimization on the first requests.
-# Must warmup similar to actual run so that all instances are ready
-$PERF_CLIENT -v -i ${PERF_CLIENT_PROTOCOL} -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
-                -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY}
+# Overload use of PERF_CLIENT_PROTOCOL for convenience with existing test and 
+# reporting structure, though "triton_c_api" is not strictly a "protocol".
+# NOTE: tritonserver will be ran in-process with perf_analyzer
+if [[ "${PERF_CLIENT_PROTOCOL}" == "triton_c_api" ]]; then
+    # Using C API requires extra info to start in-process server
+    # NOTE: PA C API doesn't expose --backend-directory, but defaults to
+    #       ${TRITON_DIR}/backends
+    # NOTE: PA C API doesn't expose --backend-config, but this is only used
+    #       for tfserving grpc test currently.
+    SERVICE_ARGS="--service-kind triton_c_api \
+                  --triton-server-directory ${TRITON_DIR} \
+                  --model-repository ${MODEL_REPO}"
+# Otherwise run server as separate process as normal via HTTP/GRPC
+else
+    SERVICE_ARGS="-i ${PERF_CLIENT_PROTOCOL}"
+
+    # Don't start separate server if testing perf with C API
+    SERVER_LOG="${NAME}.serverlog"
+    run_server
+    if (( $SERVER_PID == 0 )); then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+
+    # NOTE: Running PA for warmup doesn't make sense for C API since it uses
```

Comment:
Move these lines below L98. And reword to may be something like this:

```
Note: Running extra PA for warmup doesn't make sense for C API since it uses in-process tritonserver which will exit along with this PA process.
```